src/lib.rs

```diff
@@ -2269,11 +2046,10 @@ struct State {
 
     /// Storage of mapping from preview1 file descriptors to preview2 file
     /// descriptors.
-    ndescriptors: Cell<u16>,
-    descriptors: UnsafeCell<MaybeUninit<[Descriptor; MAX_DESCRIPTORS]>>,
-
-    /// Points to the head of a free-list of closed file descriptors.
-    closed: Option<Fd>,
+    ///
+    /// Do not use this member directly - use State::descriptors() to ensure
+    /// lazy initialization happens.
+    descriptors: Descriptors,
```

Comment:
Could consider making this `Option<Descriptors>` to discourage accidental direct use.

Reply:
Or `RefCell<Option<Descriptors>>` if necessary.  Probably overkill, though.

Reply:
Good idea, I was able to do it with RefCell<Option<Descriptors>>. I hadn't seen that you could use RefCell::try_borrow().unwrap_or_else(|| unreachable!()) to avoid pulling in panic machinery. Now the invalid state is unrepresentable.